matron/src/hardware/screen.c

```diff
@@ -529,5 +545,56 @@ void screen_surface_display_region(screen_surface_t *s,
     cairo_restore(cr);
 }
 
+screen_context_t *screen_context_new(screen_surface_t *target) {
+    cairo_surface_t *image = (cairo_surface_t *)target;
+    // NOTE: cairo_create increases the ref count on image which avoids a double
+    // free and allows the image and context to be freed in any order
+    cairo_t *context = cairo_create(image);
+    cairo_status_t status = cairo_status(context);
+    if (status == CAIRO_STATUS_SUCCESS) {
+        return (screen_context_t *)context;
+    }
+    fprintf(stderr, "context_new: %s (%d)\n", cairo_status_to_string(status), status);
```

Comment:
do you want to leave this in? (debug print?)

Reply:
yes, this one and that other one in `screen_surface_new` new where intentionally left in; it will help diagnose the source of problems with the higher up operations in weaver throw a more generic lua error.